pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -425,13 +394,28 @@ func (pc *PodController) configureSecondaryNetwork(pod *corev1.Pod, networklist
 			klog.InfoS("NetworkAttachmentDefinition is not of type 'antrea', ignoring", "NetworkAttachmentDefinition", klog.KObj(netDefCRD))
 			continue
 		}
-		if networkConfig.NetworkType != sriovNetworkType {
+		if networkConfig.NetworkType != sriovNetworkType && networkConfig.NetworkType != vlanNetworkType {
 			// same as above, if updated, we will not process the request again.
-			klog.ErrorS(err, "NetworkType not supported for Pod", "NetworkAttachmentDefinition", klog.KObj(netDefCRD), "Pod", klog.KObj(pod))
+			klog.ErrorS(err, "Secondary network type not supported", "NetworkAttachmentDefinition", klog.KObj(netDefCRD), "Pod", klog.KObj(pod))
 			continue
 		}
+		if networkConfig.NetworkType == vlanNetworkType {
+			if networkConfig.VLAN > vlanIDMax || networkConfig.VLAN < 0 {
+				klog.ErrorS(nil, "Invalid VLAN ID", "NetworkAttachmentDefinition", klog.KObj(netDefCRD), "Pod", klog.KObj(pod))
```

Comment:
nit: maybe add the VLAN value to the log

Reply:
Done